metabase/metabase
# Add caching for parameter values calls

Epic https://github.com/metabase/metabase/issues/26898

Adds caching for API calls for parameters. Previously this was only implemented for dashboards, but this PR adds a similar logic for parameters in questions (both saved and unsaved, where we call `/api/dataset/parameter` endpoints). The PR also consolidates logic for caching dashboard parameters.

Includes changes from https://github.com/metabase/metabase/pull/27995

How to test:
- New -> SQL Query -> `SELECT * FROM PRODUCTS WHERE {{tag}}`
- Field Filter -> Products -> Category
- Click on the filter widget and make sure it returns values
- Clicking again should not result in any API calls
- Change the values config for this parameter, e.g. Dropdown list -> Custom list
- Make sure that if you click on the filter widget this time, it returns the new list
- Clicking again should not results in any API calls


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27977)
